File: temp_tests_folder/test2.py
class Data:
    def __init__(self, value: str, timestamp: int):
        self.value = value
        self.timestamp = timestamp

class TimeMapPY:
    def __init__(self):
        self.map = {}  # key -> List[Data]

    def set(self, key: str, value: str, timestamp: int) -> None:
        if key not in self.map:
            self.map[key] = []
        self.map[key].append(Data(value, timestamp))

    def get(self, key: str, timestamp: int) -> str:
        if key not in self.map:
            return ""

        data = self.map[key]
        return self._binary_search(data, timestamp)

    def _binary_search(self, data: list, timestamp: int) -> str:
        l, r = 0, len(data) - 1
        res = ""

        while l <= r:
            m = (l + r) // 2
            if data[m].timestamp <= timestamp:
                res = data[m].value
                l = m + 1
            else:
                r = m - 1

        return res

File: temp_tests_folder/test2.ts
class DataTS {
    value: string;
    timestamp: number;
  
    constructor(value: string, timestamp: number) {
      this.value = value;
      this.timestamp = timestamp;
    }
  }
  
  class TimeMapTS {
    private map: Map<string, Data[]>;
  
    constructor() {
      this.map = new Map();
    }
  
    set(key: string, value: string, timestamp: number): void {
      if (!this.map.has(key)) {
        this.map.set(key, []);
      }
      this.map.get(key)!.push(new Data(value, timestamp));
    }
  
    get(key: string, timestamp: number): string {
      if (!this.map.has(key)) return "";
  
      const dataList = this.map.get(key)!;
      return this.binarySearch(dataList, timestamp);
    }
  
    private binarySearch(dataList: Data[], timestamp: number): string {
      let l = 0;
      let r = dataList.length - 1;
      let res = "";
  
      while (l <= r) {
        const m = Math.floor((l + r) / 2);
        if (dataList[m].timestamp <= timestamp) {
          res = dataList[m].value;
          l = m + 1;
        } else {
          r = m - 1;
        }
      }
  
      return res;
    }
  }
  